test/integration/targets/azure_rm_mysqlserver/tasks/main.yml

```diff
@@ -208,13 +210,44 @@
     resource_group: "{{ resource_group }}"
     server_name: mysqlsrv{{ rpfx }}
     name: testdatabase
+    collation: English_United States.1252
+    charset: UTF8
   register: output
 - name: Assert the state has not changed
   assert:
     that:
       - output.changed == false
       - output.name == 'testdatabase'
 
+- name: Try to update database without force_update
+  azure_rm_mysqldatabase:
+    resource_group: "{{ resource_group }}"
+    server_name: mysqlsrv{{ rpfx }}
+    name: testdatabase
+    collation: SQL_Latin1_General_CP1_CS_AS
+    charset: WIN1252
+  register: output
+- name: Assert the state has not changed
+  assert:
+    that:
+      - output.changed == false
```

Comment:
shouldn't the updating failed? if  no update happens?

Reply:
that's a good question. according to current documentation it should not.
I am considering adding a warning (as we discussed already regarding unchangeable properties).